### PR TITLE
Uuid Marker

### PR DESCRIPTION
**Pull request by jojolepro**
## Checklist

* [X] I've added tests for all code changes and additions (where applicable)
* [ ] I've added a demonstration of the new feature to one or more examples
* [ ] I've updated the book to reflect my changes
* [X] Usage of new public items is shown in the API docs

## API changes

Added UuidMarker and UuidMarkerAllocator.
They work the same way their U64 counterparts do.

Added "uuid_entity" feature gate.

---
This change is [Reviewable](https://reviewable.io/reviews/slide-rs/specs/584)

### CODE REVIEW

File: src/saveload/uuid.rs [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+    }
+
+    /// Get the current uuid.
+    pub fn uuid(&self) -> &Uuid {
+        &self.uuid
+    }
+
+    /// Set a new uuid.
+    pub fn set_uuid(&mut self, uuid: Uuid) {
+        self.uuid = uuid;
```
Comment (torkleyy) [inline src/saveload/uuid.rs]: I don't think we should permit this. Won't this invalidate the mapping?

File: Cargo.toml [hunk 2]
```diff
@@ -34,16 +34,20 @@ shred-derive = "0.6.0"
...
 
 serde = { version = "1.0", optional = true, features = ["serde_derive"] }
 
 [features]
 default = ["parallel"]
 parallel = ["rayon", "shred/parallel", "hibitset/parallel"]
 nightly = ["shred/nightly"]
+uuid_entity = ["uuid", "serde"]
+stdweb = ["uuid/stdweb"]
+wasm-bindgen = ["uuid/wasm-bindgen"]
```
Comment (torkleyy) [inline Cargo.toml]: Is this something we have to expose?

Comment (jojolepro) [reply Cargo.toml]: if you want to support both stdweb (cargo-web) and wasm then yes.

**Comment by torkleyy** [comment]
Thanks!
bors r+

**Pull request merged**